### PR TITLE
Additional seeding for reproducibility

### PR DESCRIPTION
**Pull request by rfriedman22**
#### Reference Issues/PRs

Fixes issues with reproducibility when random seed is provided as discussed in the [Google group](https://groups.google.com/g/selene-sdk/c/MvGGuCT0ri0)

#### What does this implement/fix? Explain your changes.
Seed the Python and Numpy random number generator with the same seed used for Torch, and set torch to use deterministic algorithms.

#### What testing did you do to verify the changes in this PR?
Added these lines to a CLI script and ran Selene twice with the same input and same seed, and verified the output of the training loss and validation performance metrics was identical.

### CODE REVIEW

**Comment by kathyxchen** [comment]
LGTM, thank you @rfriedman22 for your work :)
